core/src/main/java/org/elasticsearch/cluster/routing/allocation/decider/Decision.java

```diff
@@ -358,5 +355,14 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
             builder.endArray();
             return builder;
         }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            out.writeBoolean(true); // flag indicating it is a multi decision
```

Comment:
We could, but I didn't want to break the wire protocol for `Decision` (its the same reason I didn't reorder the ids in the `Type` enum which would've made the comparing easier).  The `Decision` class (and therefore, `Type` as well) can be serialized over the wire protocol in case of running a reroute command with explanations turned on.  `RoutingExplanations` relies on serializing the `Decision` class.  I don't think the benefit outweighs breaking the wire protocol here, even though it may be a rare case.  WDYT?